Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py

```diff
@@ -3484,38 +3496,48 @@ def add_modified_remote_offenses(client: Client,
         print_context_data_stats(new_context_data, "Get Modified Remote Data - Before update")
         mirrored_offenses_queries = context_data.get(MIRRORED_OFFENSES_QUERIED_CTX_KEY, {})
         finished_offenses_queue = context_data.get(MIRRORED_OFFENSES_FINISHED_CTX_KEY, {})
+        current_concurrent_searches = get_concurrent_searches(context_data)
+        offense_ids_to_search = []
 
-        # Query only the first offense_ids, to avoid timeouts in this function.
-        top_offense_ids = set(sorted([int(offense_id) for offense_id in mirrored_offenses_queries])[:MAX_SEARCHES_QUEUE])
-        top_queries = {offense_id: search_id for offense_id, search_id in mirrored_offenses_queries.items()
-                       if int(offense_id) in top_offense_ids}
-        for offense_id, search_id in top_queries.items():
+        for offense_id, search_id in mirrored_offenses_queries.copy().items():
             if search_id in {QueryStatus.WAIT.value, QueryStatus.ERROR.value}:
                 # if search_id is waiting or error, we will try to search again
-                search_id = create_events_search(client, fetch_mode, events_columns, events_limit, int(offense_id))
-                mirrored_offenses_queries[offense_id] = search_id
-                changed_ids_ctx.append(offense_id)
+                offense_ids_to_search.append(offense_id)
+                continue
             # If the search finished, move it to finished queue
             _, status = poll_offense_events(client, search_id, should_get_events=False, offense_id=int(offense_id))
             if status == QueryStatus.ERROR.value:
                 time.sleep(FAILURE_SLEEP)
                 print_debug_msg(f'offense {offense_id}, search query {search_id}, status is {status}')
                 mirrored_offenses_queries[offense_id] = QueryStatus.ERROR.value
+                current_concurrent_searches -= 1
+
             elif status == QueryStatus.SUCCESS.value:
                 del mirrored_offenses_queries[offense_id]
                 finished_offenses_queue[offense_id] = search_id
                 # add the offense id to modified in order to run get_remote_data
                 new_modified_records_ids.add(offense_id)
                 changed_ids_ctx.append(offense_id)
+                current_concurrent_searches -= 1
             else:
                 print_debug_msg(f'offense {offense_id}, search query {search_id}, status is {status}')
-            time.sleep(EVENTS_MODIFIED_SECS)
+
+        for offense_id in offense_ids_to_search:
+            if current_concurrent_searches >= MAX_SEARCHES_QUEUE:
+                break
```

Comment:
Add (info?) log message that MAX_SEARCHES was reached, and not creating another search.

Reply:
added